src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -79,225 +124,459 @@ const useStyles = makeStyles()(() => ({
+                            </FormHelperText>
+                          </Box>
+                        </Grid>
+                        <Grid item xs={12} lg={4}>
+                          <Box
+                            className={classes.boxBottom}
+                            data-testid="BoxBottom"
+                          >
+                            <InputLabel id="frequency-label">
+                              {t('Frequency')}
+                            </InputLabel>
+                            <Select
+                              className={classes.select}
+                              inputProps={{
+                                'data-testid': 'pledgeFrequency-input',
+                              }}
+                              data-testid="pledgeFrequency"
+                              label={t('Frequency')}
+                              labelId="frequency-label"
+                              placeholder="Frequency"
+                              size="small"
+                              style={{ width: '100%' }}
+                              value={pledgeFrequency}
+                              onChange={(event) =>
+                                setFieldValue(
+                                  'pledgeFrequency',
+                                  event.target.value,
+                                )
+                              }
+                            >
+                              <MenuItem value="Frequency" disabled>
+                                {t('Frequency')}
+                              </MenuItem>
+                              {Object.entries(frequencies).map(
+                                ([freqValue, freqTranslated]) => (
+                                  <MenuItem
+                                    value={freqValue}
+                                    key={freqValue}
+                                    data-testid="pledgeFrequencyOptions"
+                                  >
+                                    {freqTranslated}
+                                  </MenuItem>
+                                ),
+                              )}
+                            </Select>
+                            <FormHelperText
+                              error={true}
+                              data-testid="pledgeFrequencyError"
+                            >
+                              {errors.pledgeFrequency && errors.pledgeFrequency}
+                            </FormHelperText>
+                          </Box>
+                        </Grid>
+                      </Grid>
+                    </Grid>
+                    {!!donations.length && (
+                      <Grid
+                        container
+                        className={classes.donationsTable}
+                        lg={12}
+                      >
+                        {donations.map((donation) => (
```

Comment:
Right now the donations show the most recent date on the left hand side. So it goes from newest to oldest. Can we try `donations.reverse()` ?